Scribble scroll fix

mtallenca opened the pull request:
## Description

The scribble area overlays the editor area.  This fix keeps the scribble area from scrolling as the editor scrolls.  Without this fix, the scribble area can scroll off the screen.

- No, this is *not* a breaking change.

singerdmx commented:
@mtallenca Can you fix this issue?

`   info • 'isShiftPressed' is deprecated and shouldn't be used. Use HardwareKeyboard.instance.isShiftPressed instead. This feature was deprecated after v3.18.0-2.0.pre • lib/src/widgets/raw_editor/raw_editor_state.dart:908:41 • deprecated_member_use`

singerdmx commented:
@mtallenca I may have to revert this if it cannot be fixed

https://github.com/singerdmx/flutter-quill/actions/runs/7937269064/job/21674160453

ellet0 commented:
> @mtallenca I may have to revert this if it cannot be fixed
> 
> https://github.com/singerdmx/flutter-quill/actions/runs/7937269064/job/21674160453

This because of flutter 3.19, they have new changes and they deprecated some methods, [for more info](https://docs.flutter.dev/release/release-notes/release-notes-3.19.0)

The CI automatically use the latest version for Flutter, it should have nothing to do with this PR, the solution is to update the project to work with 3.19, by the time this PR GitHub action run the latest stable release is 3.19

mtallenca commented:
Taking a look at it now...

On Fri, Feb 16, 2024 at 3:14 PM Ellet Hnewa ***@***.***>
wrote:

> @mtallenca <https://github.com/mtallenca> I may have to revert this if it
> cannot be fixed
>
>
> https://github.com/singerdmx/flutter-quill/actions/runs/7937269064/job/21674160453
>
> This because of flutter 3.19, they have new changes and they deprecated
> some methods, for more info
> <https://docs.flutter.dev/release/release-notes/release-notes-3.19.0>
>
> The CI automatically use the latest version for Flutter, it should have
> nothing to do with this PR, the solution is to update the project to work
> with 3.19, by the time this PR GitHub action run the latest stable release
> is 3.19
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/singerdmx/flutter-quill/pull/1745#issuecomment-1949468213>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXMGLMBA6IDXZMKAGXHJQ3YT7R6ZAVCNFSM6AAAAABDLEIYD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNBZGQ3DQMRRGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ellet0 commented:
> Taking a look at it now...
> […](#)
> On Fri, Feb 16, 2024 at 3:14 PM Ellet Hnewa ***@***.***> wrote: @mtallenca <https://github.com/mtallenca> I may have to revert this if it cannot be fixed https://github.com/singerdmx/flutter-quill/actions/runs/7937269064/job/21674160453 This because of flutter 3.19, they have new changes and they deprecated some methods, for more info <https://docs.flutter.dev/release/release-notes/release-notes-3.19.0> The CI automatically use the latest version for Flutter, it should have nothing to do with this PR, the solution is to update the project to work with 3.19, by the time this PR GitHub action run the latest stable release is 3.19 — Reply to this email directly, view it on GitHub <[#1745 (comment)](https://github.com/singerdmx/flutter-quill/pull/1745#issuecomment-1949468213)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAXMGLMBA6IDXZMKAGXHJQ3YT7R6ZAVCNFSM6AAAAABDLEIYD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNBZGQ3DQMRRGM> . You are receiving this because you were mentioned.Message ID: ***@***.***>

If you want, please take a look at [this](https://github.com/singerdmx/flutter-quill/pull/1746)

I already updated the deprecated code but I have not tested the functionalities of the updated code, it should work fine because it's not a breaking change, however I'm not sure if the new api is available in 3.16, if not then we need to update the minimum version of flutter to 3.19

mtallenca commented:
I looked at
 flutter/packages/flutter/lib/src/services/hardward_keyboard.dart - the new
HardwareKeyboard.instance.isControlPressed (and others) were added to
master channel Oct 27.  This commit is not in 3.16 - you'll need to update
the minimum sdk to 3.19.

commit 5907c9757e3f23857ab306250a707400dd200277
Author: Greg Spencer ***@***.***>
Date:   Fri Oct 27 11:52:08 2023 -0700
    Add `isLogicalKeyPressed` to `KeyEvent` (#136856)
    ## Description
    Adds some convenience methods to `KeyEvent` that allow testing to see
if a logical or physical key is pressed from the event object. These are
similar to the ones already on `RawKeyEvent`, and will make migration the
to `KeyEvent` easier (so it could more easily be a `flutter fix` migration).

    Added:
    - `bool isLogicalKeyPressed(LogicalKeyboardKey key)`
    - `bool isPhysicalKeyPressed(PhysicalKeyboardKey key)`
    - `bool get isControlPressed`
    - `bool get isShiftPressed`
    - `bool get isAltPressed`
    - `bool get isMetaPressed`

    ## Related Issues

     - https://github.com/flutter/flutter/issues/136419

    ## Tests

     - Added tests for the new methods.

Message ID: ***@***.***>
>

mtallenca commented:
We are running those same changes in our app built with 3.19  yesterday and
not seeing any errors.

On Fri, Feb 16, 2024 at 4:31 PM Michael Allen ***@***.***> wrote:

> I looked at
>  flutter/packages/flutter/lib/src/services/hardward_keyboard.dart - the new
> HardwareKeyboard.instance.isControlPressed (and others) were added to
> master channel Oct 27.  This commit is not in 3.16 - you'll need to update
> the minimum sdk to 3.19.
>
> commit 5907c9757e3f23857ab306250a707400dd200277
> Author: Greg Spencer ***@***.***>
> Date:   Fri Oct 27 11:52:08 2023 -0700
>     Add `isLogicalKeyPressed` to `KeyEvent` (#136856)
>     ## Description
>     Adds some convenience methods to `KeyEvent` that allow testing to see
> if a logical or physical key is pressed from the event object. These are
> similar to the ones already on `RawKeyEvent`, and will make migration the
> to `KeyEvent` easier (so it could more easily be a `flutter fix` migration).
>
>     Added:
>     - `bool isLogicalKeyPressed(LogicalKeyboardKey key)`
>     - `bool isPhysicalKeyPressed(PhysicalKeyboardKey key)`
>     - `bool get isControlPressed`
>     - `bool get isShiftPressed`
>     - `bool get isAltPressed`
>     - `bool get isMetaPressed`
>
>     ## Related Issues
>
>      - https://github.com/flutter/flutter/issues/136419
>
>     ## Tests
>
>      - Added tests for the new methods.
>
> Message ID: ***@***.***>
>>
>

mtallenca commented:
raw_editor_state.dart is using the new 3.19 methods.

i.e. line 799

  KeyEventResult _onKeyEvent(node, KeyEvent event) {

    // Don't handle key if there is a meta key pressed.

    if (HardwareKeyboard.instance.isAltPressed ||

        HardwareKeyboard.instance.isControlPressed ||

        HardwareKeyboard.instance.isMetaPressed) {

      return KeyEventResult.ignored;

    }

On Fri, Feb 16, 2024 at 4:46 PM Ellet Hnewa ***@***.***>
wrote:

> HardwareKeyboard.instance.isControlPressed
>
> But we are not using HardwareKeyboard.instance.isControlPressed in the
> project, should we still require the project to use 3.19?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/singerdmx/flutter-quill/pull/1745#issuecomment-1949543724>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXMGLI5QIN4SZGZEJBGU7DYT74UVAVCNFSM6AAAAABDLEIYD6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSNBZGU2DGNZSGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

ellet0 commented:
I just updated the code, and require 3.19 as minimum to use the latest version of flutter quill

Also you might want to look at the PR I sent along with the new commit which is on the master branch and not with the PR, which has some fixes
